mdp/linear_flows_online.py

```diff
@@ -0,0 +1,607 @@
+from builtins import str
+import mdp
+from mdp import numx
+from .linear_flows import FlowException, FlowExceptionCR, _sys, _traceback
+from collections import deque as _deque
+
+
+class OnlineFlowException(mdp.MDPException):
+    """Base class for exceptions in Flow subclasses."""
+    pass
+
+
+class OnlineFlow(mdp.Flow):
+    """An 'OnlineFlow' is a sequence of online/executable nodes that are trained and executed
+    together to form a more complex algorithm.  Input data is sent to the
+    first node and is successively processed by the subsequent nodes along
+    the sequence.
+
+    Using an online flow as opposed to handling manually a set of nodes has a
+    clear advantage: The general online flow implementation automatizes the
+    training (including supervised training and multiple training phases),
+    execution, and inverse execution (if defined) of the whole sequence.
+
+    OnlineFlow sequence can contain trained or non-trainable Nodes and optionally
+    a terminal trainable Node.
```

Comment:
I think we generally have to be a little careful with the term "training" and "trainable". On the one hand there are "online" nodes that "learn" in their executable state (i.e., with node.execute). On the other hand there are "trainable" nodes, that have standard training phases.
Here's my current understanding (please correct me if this is off): What OnlineFlow really does is to enable specific additional data (supervision signals) to be passed to each node in the flow during the training of the last node in the flow (that is allowed to be trainable).